tensorflow_datasets/core/download/downloader.py

```diff
@@ -168,10 +159,15 @@ def _sync_download(self, url, destination_path):
     session = requests.Session()
     if _DRIVE_URL.match(url):
       url = self._get_drive_url(url, session)
-    response = session.get(url, stream=True)
-    if response.status_code != 200:
-      raise DownloadError(
-          'Failed to get url %s. HTTP code: %d.' % (url, response.status_code))
+    if not url.startswith('ftp'):
+      response = session.get(url, stream=True)
+      if response.status_code != 200:
```

Comment:
can the error check be moved below the if/else so that it's used in both cases?

Reply:
Actually That's not possible. I tried it at first.
I'm using `urllib.request` for FTP service. And surprisingly enough, even after successfully opening the stream, the `code` attribute of the ioBufferedReader optained is `None`. The HTTP Status Codes are put after the whole file is downloaded. But by that time, We don't need the status code.
But, There's a plus point to this issue. If there's any issue, `urllib.error.URLError` is raised, stopping the execution.

Reply:
@rsepassi